ls1intum/Artemis
# [Database] StudentScores and TutorScores

### Checklist

### Motivation and Context
To make the leaderboards for my bachelor thesis more efficent we want to have the scores of the students closer with the exercises themselves. In the future we also want to get rid of the views that are currently used for the tutor leaderboards and the tutor dashboards.
Also see Issue #2044.

### Description
There will be two new tables added to the database. The student_score table will include the most recent result and score of an exercise for a student. The tutor_score table is an aggregation of the five views that are currently used for the tutor leaderboards and the dashboard.
The appropriate updates for the tables so that the data is correct and up to date will come later.

### Steps for Testing
best tested locally because currently there are no changes that can be seen or are sent to the client
1. Create exercises and participate in them with different users.
2. Assess the submission.
3. Have a look in your db and check if all StudentScores and TutorScores are entered correctly.
4. Change assessments and complain or ask for more feedback and answer those requests.
5. Step 3.

### Test Coverage
- StudentScore: 100%
- StudentScoreResource: 100%
- StudentScoreService: 96%
- TutorScore: 96%
- TutorScoreResource: 100%
- TutorScoreService: 87%
